### DISCUSSION TITLE
multiple reactions in TGA

### DISCUSSION THREAD

**Discussion by SaharRasoulpour** [firemodels/fds#12699] [Q&A]
Hello,

I have a question regarding multiple reactions in a TGA sample. All input values are identical. When I run these three reactions in the TGA sample input file, there is a small deviation in the offset part. Would it be possible for you to review my input file to ensure that all of my reactions are implemented correctly? What is your suggestion regarding this issue? 

![1](https://github.com/firemodels/fds/assets/143542094/298e27b3-c30d-48d5-ac01-2ce4be5d2d99)
[tga_sample.txt](https://github.com/firemodels/fds/files/14791001/tga_sample.txt)

**Comment by rmcdermo** [comment]
Can you please clarify your question.  I don't know what "All" and "MB97" refer to and how to connect this with your input file.  And I'm confused by your comment "All input values are identical".  I suggest creating two separate input files: one for the blue curve and one for the green.

**Comment by SaharRasoulpour** [comment]
The input parameters, such as density, conductivity, heating rate, A, E, mass fraction, and … are identical and specified in the paper I want to compare my results with. I've labeled my data as "All." Additionally, MB97 represents values from another paper. I aim to maintain consistency by fixing my normalized mass with MB97, ensuring identity since the input values are the same (I derived my initial and input values from MB97 in that paper).

**Comment by mcgratta** [comment]
I have never used simultaneous reactions for a material, that is having multiple sets of kinetic constants for a single material. Are you comparing your results to someone else who has used FDS? If not, there might be subtle differences in the way the reaction rates are written by the two codes.

**Reply by SaharRasoulpour** [reply]
I'm not comparing my results with the FDS results. Instead, I'm trying to compare them with the results from GPyro.

**Comment by TristanHehnen** [comment]
Hi!

It is not quite clear to me if your input file is about  "All" or "MB97 ". I guess it is "All", as I understand it to be your model. 

- From the schematic in the plot, it seems to me that there is a decomposition reaction missing in your input file: R4. 

- In your input file, 'TAR' is defined as a `SPEC`, which will be a gas. It seems you would like the material to transform into tar and then decompose into a gas. I would model the tar as `MATL` that further decomposes into gases, at least from how I understand the schematic in the plot. 

- In the definitions of the intermediate materials, e.g. 'CELLULOSE1', there are two `SPEC` being released and one `MATL`, with the `NU_SPEC(1,1)=1`, `NU_SPEC(1,2)=0.35` and `NU_MATL(1,2)=0.65` I'm not sure how FDS deals with this situation. I would suggest to make sure all these yields add up to 1, see also "9.2.1 Reaction Mechanism".

- For the `MATL` 'VIRGIN_CELLULOSE', 'VIRGIN_HEMICELLULOSE' and 'VIRGIN_LIGNIN' you've specified `REFERENCE_TEMPERATURE` as well as `A` and `E`. Please check the users guide, section "9.2.2 Reaction Rates" (p. 93). FDS uses the `REFERENCE_TEMPERATURE` to determine `A` and `E`, thus you don't need it here since you apparently already have `A` and `E`.

- With the comments from above, I would suggest to implement the decomposition scheme as follows (NU_MATL/NU_SPEC values are examples, choose what ever is appropriate for the given model to add up to 1):

```
&MATL ID = 'Component01_VIRGIN',
[Reaction R1],
NU_MATL(1,1) = 0.42,
MATL_ID(1,1) = 'Component01A_ACTIVE',
NU_MATL(1,2) = 0.58,
MATL_ID(1,2) = 'Component01B_ACTIVE' /

&MATL ID = 'Component01A_ACTIVE',
[Reaction R2],
NU_MATL(1,1) = 1.0,
MATL_ID(1,1) = 'Component01_TAR' /

&MATL ID = 'Component01B_ACTIVE',
[Reaction R3],
NU_MATL(1,1) = 0.25,
MATL_ID(1,1) = 'Component01_CHAR',
NU_SPEC(1,2) = 0.75,
SPEC_ID(1,2) = 'Component01_GAS' /

&MATL ID = 'Component01_TAR',
[Reaction R4],
NU_SPEC(1,1) = 1.0,
SPEC_ID(1,1) = 'Component01_GAS' /

&MATL ID = 'Component01_CHAR',
[inert solid residue] /

&SPEC ID = 'Component01_GAS',
[some gaseous species; combustible?] /
```

- Be careful when tracking all the different materials and species to determine/plot the residual mass. For example consider a model where one `MATL` has a reaction that completely transforms it into another `MATL`. If you track the first `MATL` alone it will vanish. However, at the same time the second `MATL` is created. Thus, the total sample mass stays the same.

- You may also want to compare your model to DCS data, to make sure that your specific heat capacities and heats of reactions work as intended.

Hope this helps.

**Comment by drjfloyd** [comment]
@TristanHehnen 

"In the definitions of the intermediate materials, e.g. 'CELLULOSE1', there are two SPEC being released and one MATL, with the NU_SPEC(1,1)=1, NU_SPEC(1,2)=0.35 and NU_MATL(1,2)=0.65 I'm not sure how FDS deals with this situation. I would suggest to make sure all these yields add up to 1, see also "9.2.1 Reaction Mechanism".

There are two parallel reactions for this material. The first reaction just makes gas. The second reaction makes gas and a residue material. Mass should be conserved as some of the material will react via the first reaction and some the second reaction but none of the material should be reacting by both reactions at once.

@SaharRasoulpour 

You could do this reaction scheme by hand in a spreadsheet. You can make a finely resolved column of time and temperature, start with the virgin material, compute all reaction rates, and do an explicit update of masses for the various materials.  That could help establish which of the two curves is correct.

**Reply by TristanHehnen** [reply]
@drjfloyd I understand the intention to have two parallel reactions, but the sum of the three `NU_*` is larger than 1. Aren't the  `NU_*` mass fractions of the initial `MATL`? 

So I guess my question here would be if FDS will normalise all the  `NU_*` and in this case here (since the sum is 2) really NU_MATL(1,2)=0.65/2 is produced, thus leading to less residue than expected.

**Reply by TristanHehnen** [reply]
Otherwise, how does FDS know how to distribute the initial mass between the two reaction? 
This is what I'm not clear about.

**Reply by drjfloyd** [reply]
Each reaction has its own A and E. At any time in the simulation the current solid temperature plus the A and E give a reaction rate for the kg/s of the material that reacts via each reaction. So if at time t for CELLULOSE1 (for example) we have reaction rate 1 = 0.1 kg/s and reaction rate 2 = 0.3 kg/s then CELLULOSE1 will decrease by 0.4 kg/s (the sum of the two reaction rates) and produce 0.1 * 1 + 0.3 * 0.65 = 0.295 kg/s of NITROGEN and 0.3 x 0.35 = 0.105 kg/s of residue and 0.295 +0.105 = 0.4 kg/s

**Reply by TristanHehnen** [reply]
Thank you for the clarification! 
I think, I did not express well what I meant. 



Let's assume we want to run a TGA simulation where: 
- the sample has two distinct decomposition reactions 
- we can attribute both reactions to a specific component of the sample and we know that both make up half of the sample mass each
- the first reaction will produce only gas ("first" as in "happens at lower temperature")
- the second reaction will produce 50% gas and 50% inert solid residue.

We can define then two `MATL` with the same thermophysical properties and different decomposition reactions. This limits each reactions access to the initial sample material and leaves the expected residue of 25% of the original sample mass.  

We can also define a `MATL` that contains both reactions. However, then FDS does not know which reaction should consume which portion of the sample. With this, each reaction has access to the full sample material.  In this case, which ever reaction happens at lower temperatures will consume (possibly) the whole sample material for itself. It is a race between the different reactions then.

I've set up a brief demonstration with different decomposition schemes (FDS-6.9.0-0-g6339569-master):
1. "2 Materials" has the two distinct `MATL`, one for each reaction
2. "2 Reactions"  has a single  `MATL`, containing both reaction
3. "2 Reactions, flipped" has a single  `MATL`, containing both reaction but in reverse order such that the residue producing reaction happens at lower temperature

Case 1 is the desired behaviour for the example here.
One can see that in case 2 all the material is consumed during the first reaction, thus no residue is produced.
In case 3 however, half of the sample mass is left as residue, because this time the reaction that produces residue comes first.


![grafik](https://github.com/firemodels/fds/assets/34650649/8b3fde2e-e3ed-44f5-8d75-250c6b6808b0)


![grafik](https://github.com/firemodels/fds/assets/34650649/a3199bea-0fc1-4342-b77f-3435237d61f3)


[TwoMaterials.txt](https://github.com/firemodels/fds/files/14811487/TwoMaterials.txt)
[TwoReactions.txt](https://github.com/firemodels/fds/files/14811488/TwoReactions.txt)
[TwoReactionsFlipped.txt](https://github.com/firemodels/fds/files/14811489/TwoReactionsFlipped.txt)

I think that this is also the answer to the original question here. 
Due to setting up multiple reactions in a single `MATL` they compete for the same supply and therefore the expected residue masses are off.

**Reply by TristanHehnen** [reply]
Also I did mix up the `NU_*` they are not related to the initial sample mass but the reaction. I'm typically using only `MATL` definitions with single decomposition reactions, in which case `NU_*` appear as if they were related to the initial mass.

**Reply by drjfloyd** [reply]
Yes, parallel reactions make it hard to know just by looking at the A and E values how much of each reaction will occur. However, if these A and E were developed from a TGA test at the same heating rate used in the model in a manner consistent with the reaction rate equation used in FDS and Gpyro, then you would have the ability in that process to tune the A and E for whatever residue is desired. You would not expect the same end state for a different heating rate as the time spent at each temperature would change.

Based on comments from the OP, the two curves are FDS and Gpyro. These are using the same equation for reaction rate.  If both cases are at the same heating rate, and if both cases have the same A,E,rho, etc. (which isn't clear if that is the case see my comment below from yesterday), then would expect very similar results between the two. You might be correct and it may just be slight differences in numerics between FDS and Gpyro that are resulting in divergence because of the parallel reactions. But first I think need to make sure that the two inputs are identical.

**Comment by SaharRasoulpour** [comment]
Thank you for your comment and suggestion. Reactions R1–R3 are considered primary reactions, while R4 is classified as a secondary reaction. The primary reactions primarily focus on converting the initial species into an active state, which subsequently decomposes into lower molecular weight gases and char. On the other hand, the secondary reaction, R4, is responsible for the conversion of tar (gas) into low molecular weight gases. Here, Reaction 4 doesn't significantly impact the results, so I excluded it from my input file. When I remove the reference temperature, I noticed that my plot slightly deviates below the corresponding data, similar to a linear plot.

**Comment by drjfloyd** [comment]
You are overspecifying your inputs. FDS uses REFERENCE_TEMPERATURE and HEATING_RATE to obtain A and E.  In this case the temperature and rate are overriding the A and E you have specifiied (look at the .out file to see what A and E results).  Which inputs did the gpyro model use? Your E appears to have the wrong units. 

      REFERENCE_TEMPERATURE     = 620
      HEATING_RATE              = 5
      N_REACTIONS               = 1
      A(1)                      = 2.8E19
      E(1)                      = 242.4

You conductivity values appear to have the wrong units.  Wood is is on the order of 0.1 W/m/K. Your input is 1E-4.

**Reply by drjfloyd** [reply]
I ran this case using your original file and then modified to use only the A and E values (correcting for the factor of 1000 unit error).  The results are shown below.  It appears in addition to errors in the input file that you made an error in plotting. The FDS temperautre output is in deg C and you plotted as deg K. This also suggests that your REFEERENCE_TEMPERATURE input has a units error.  After those changes the end residue is at 33 % which looks to be close to the MB97 value.  The curve shape also looks similar. Both the modified results and the MB97 results show a slight bumpout in the curve between a normalized mass of 0.9 and 0.8 and both reach a normalized mass of 0.5 at a little over 600 K.

<img width="498" alt="image" src="https://github.com/firemodels/fds/assets/12799217/9533ccee-9cdf-4235-ab7d-f4a4eeab6989">


[tga_sample_ae.txt](https://github.com/firemodels/fds/files/14811787/tga_sample_ae.txt)

**Reply by SaharRasoulpour** [reply]
Thank you for your feedback; 
your comment was useful to me. I noticed that you removed the heating rate from the input file. If I want to set the heating rate to 20 K/min, should I update it directly in the source file, or specify in the input file that the heating rate is 20 K/min?

**Reply by drjfloyd** [reply]
You should only specify A + E using one method.  Either use REFERENCE_TEMPERATURE, PYROLYSIS_WIDTH, and HEATING_RATE or use A and E.  I used the A and E values you had (correcting the units on E) which meant delteing the redundant inputs.  Note that HEATING_RATE should be the rate that was used in the TGA test from which the REFERENCE_TEMPERATURE and PYROLYSIS_WIDTH was taken on not some arbirtrary value.
